src/common/types.ts

```diff
@@ -48,57 +55,17 @@ export interface TransactionList {
   [currencyCode: string]: EdgeTransaction[]
 }
 
-export class WalletLocalData {
-  blockHeight: number
-  lastAddressQueryHeight: number
-  lastTransactionQueryHeight: { [currencyCode: string]: number }
-  lastTransactionDate: { [currencyCode: string]: number }
-  publicKey: string
-  totalBalances: { [currencyCode: string]: string }
-  lastCheckedTxsDropped: number
-  numUnconfirmedSpendTxs: number
-  numTransactions: { [currencyCode: string]: number }
-  otherData: unknown
-
-  constructor(jsonString: string | null) {
-    this.blockHeight = 0
-    const totalBalances: { [currencyCode: string]: string } = {}
-    this.totalBalances = totalBalances
-    this.lastAddressQueryHeight = 0
-    this.lastTransactionQueryHeight = {}
-    this.lastTransactionDate = {}
-    this.lastCheckedTxsDropped = 0
-    this.numUnconfirmedSpendTxs = 0
-    this.numTransactions = {}
-    this.otherData = undefined
-    this.publicKey = ''
-    if (jsonString !== null) {
-      const data = JSON.parse(jsonString)
+export const asWalletLocalData = asObject({
+  blockHeight: asSafeOptional(asNumber, 0),
+  lastAddressQueryHeight: asSafeOptional(asNumber, 0),
+  lastTransactionQueryHeight: asSafeOptional(asMap(asNumber), {}),
+  lastTransactionDate: asSafeOptional(asMap(asNumber), {}),
+  publicKey: asSafeOptional(asString, ''),
+  totalBalances: asSafeOptional(asMap(asString), {}),
+  lastCheckedTxsDropped: asSafeOptional(asNumber, 0),
+  numUnconfirmedSpendTxs: asSafeOptional(asNumber, 0),
+  numTransactions: asSafeOptional(asMap(asNumber), {}),
+  otherData: asSafeOptional(asUnknown, {})
```

Comment:
All of these should become `asMaybe`, except the last one, which should be `asOptional(asUnknown, {})` to get the right fallback logic. This edge case is the only difference between `asMaybe` and `asSafeOptional`.

Reply:
Crud. Didn't realize that was asMaybe. thx